Common/include/option_structure.hpp

```diff
@@ -2107,57 +2107,62 @@ static const MapType<string, ENUM_INPUT_REF> Input_Ref_Map = {
-  PERIODIC_PRIM_GG    =  8,  /*!< \brief Primitive gradient communication for Green-Gauss (periodic only). */
-  PERIODIC_SOL_LS     =  9,  /*!< \brief Solution gradient communication for weighted Least Squares (periodic only). */
-  PERIODIC_PRIM_LS    = 10,  /*!< \brief Primitive gradient communication for weighted Least Squares (periodic only). */
-  PERIODIC_LIM_SOL_1  = 11,  /*!< \brief Solution limiter communication phase 1 of 2 (periodic only). */
-  PERIODIC_LIM_SOL_2  = 12,  /*!< \brief Solution limiter communication phase 2 of 2 (periodic only). */
-  PERIODIC_LIM_PRIM_1 = 13,  /*!< \brief Primitive limiter communication phase 1 of 2 (periodic only). */
-  PERIODIC_LIM_PRIM_2 = 14,  /*!< \brief Primitive limiter communication phase 2 of 2 (periodic only). */
-  PERIODIC_IMPLICIT   = 15,  /*!< \brief Implicit update communication to ensure consistency across periodic boundaries. */
-  PERIODIC_SOL_ULS    = 16,  /*!< \brief Solution gradient communication for unwieghted Least Squares (periodic only). */
-  PERIODIC_PRIM_ULS   = 17   /*!< \brief Primitive gradient communication for unweighted Least Squares (periodic only). */
+  PERIODIC_NONE       ,  /*!< \brief No periodic communication required. */
+  PERIODIC_VOLUME     ,  /*!< \brief Volume communication for summing total CV (periodic only). */
+  PERIODIC_NEIGHBORS  ,  /*!< \brief Communication of the number of neighbors for centered schemes (periodic only). */
+  PERIODIC_RESIDUAL   ,  /*!< \brief Residual and Jacobian communication (periodic only). */
+  PERIODIC_LAPLACIAN  ,  /*!< \brief Undivided Laplacian communication for JST (periodic only). */
+  PERIODIC_MAX_EIG    ,  /*!< \brief Maximum eigenvalue communication (periodic only). */
+  PERIODIC_SENSOR     ,  /*!< \brief Dissipation sensor communication (periodic only). */
+  PERIODIC_SOL_GG     ,  /*!< \brief Solution gradient communication for Green-Gauss (periodic only). */
+  PERIODIC_PRIM_GG    ,  /*!< \brief Primitive gradient communication for Green-Gauss (periodic only). */
+  PERIODIC_SOL_LS     ,  /*!< \brief Solution gradient communication for weighted Least Squares (periodic only). */
+  PERIODIC_PRIM_LS    ,  /*!< \brief Primitive gradient communication for weighted Least Squares (periodic only). */
+  PERIODIC_SOL_ULS    ,  /*!< \brief Solution gradient communication for unwieghted Least Squares (periodic only). */
+  PERIODIC_PRIM_ULS   ,  /*!< \brief Primitive gradient communication for unweighted Least Squares (periodic only). */
+  PERIODIC_SOL_GG_R   ,  /*!< \brief Same but reconstruction. */
+  PERIODIC_PRIM_GG_R  ,  /*!< \brief Same but reconstruction. */
+  PERIODIC_SOL_LS_R   ,  /*!< \brief Same but reconstruction. */
+  PERIODIC_PRIM_LS_R  ,  /*!< \brief Same but reconstruction. */
+  PERIODIC_SOL_ULS_R  ,  /*!< \brief Same but reconstruction. */
+  PERIODIC_PRIM_ULS_R ,  /*!< \brief Same but reconstruction. */
```

Comment:
These are needed to target the reconstruction gradient during periodic communications, otherwise not only was it not communicated, but it also corrupted the "normal" gradient for viscous and source terms.